File: lib/models/budget.dart
class Budget {
  final String id;
  final String userId;
  final double amount;
  final String currency;
  final String period;
  final double alertThreshold;
  final bool isActive;
  final DateTime createdAt;
  final DateTime updatedAt;

  Budget({
    required this.id,
    required this.userId,
    required this.amount,
    required this.currency,
    required this.period,
    required this.alertThreshold,
    required this.isActive,
    required this.createdAt,
    required this.updatedAt,
  });
}

File: backend/src/main/java/com/walletapp/demo/entities/Budget.java
package com.walletapp.demo.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;

import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

/**
 * @author Briceida Fernandez
 * @version 1.0
 * 
 * Entity that represents a Budget.
 */
@Entity
@Inheritance(strategy = InheritanceType.JOINED)
@Table(name = "budgets")
@AllArgsConstructor
@NoArgsConstructor
@Setter
@Getter
public class Budget {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "budget_id")
    private Long id;

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Column(nullable = false)
    private double amount;

    @Column(nullable = false)
    private String currency;
    
    private String period;
    private double alertThreshold;
    private boolean isActive;

    @CreationTimestamp
    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
}
